auth_api_key/models/auth_api_key.py

```diff
@@ -1,38 +1,85 @@
 # Copyright 2018 ACSONE SA/NV
 # License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
 
-from odoo import api, models, tools, _
+from odoo import api, fields, models, tools, _
 
-from odoo.addons.server_environment import serv_config
 from odoo.tools import consteq
 
 from odoo.exceptions import ValidationError, AccessError
 
 
-class AuthApiKey(models.TransientModel):
+class AuthApiKey(models.Model):
     _name = "auth.api.key"
-    _description = "API Key Retriever"
+    _inherit = "server.env.mixin"
+    _description = "API Key"
+
+    name = fields.Char(required=True)
+    key = fields.Char(required=True)
```

Comment:
Right, since this field is required, I suggest some help for users who don't read the documentation. :)

help="The API key. Enter a dummy value in this field if it is obtained from the server environment configuration."